airbyte-integrations/connectors/destination-mssql-v2/src/main/kotlin/io/airbyte/integrations/destination/mssql/v2/MSSQLQueryBuilder.kt

```diff
@@ -0,0 +1,329 @@
+                    airbyteMeta.changes.add(
+                        AirbyteRecordMessageMetaChange()
+                            .withField(field.name)
+                            .withChange(AirbyteRecordMessageMetaChange.Change.NULLED)
+                            .withReason(
+                                AirbyteRecordMessageMetaChange.Reason
+                                    .DESTINATION_SERIALIZATION_ERROR
+                            )
+                    )
+                } catch (e: IllegalArgumentException) {
+                    statement.setNull(stmntIdx, sqlType)
+                    airbyteMeta.changes.add(
+                        AirbyteRecordMessageMetaChange()
+                            .withField(field.name)
+                            .withChange(AirbyteRecordMessageMetaChange.Change.NULLED)
+                            .withReason(
+                                AirbyteRecordMessageMetaChange.Reason
+                                    .DESTINATION_SERIALIZATION_ERROR
+                            )
+                    )
+                }
+                //                when (sqlType) {
+                //                    Types.LONGVARCHAR -> statement.setString(statementIndex,
+                // Jsons.serialize(value))
+                //                    Types.BOOLEAN -> statement.setBoolean(statementIndex, value as
+                // Boolean)
+                //                    Types.DATE -> statement.setDate(statementIndex, value as Date)
+                //                    Types.BIGINT ->
+                //                        statement.setLong(statementIndex, (value as
+                // BigInteger).toLong())
+                //                    Types.DECIMAL ->
+                //                        statement.setDouble(statementIndex, (value as
+                // BigDecimal).toDouble())
+                //                    Types.VARCHAR -> statement.setString(statementIndex, value as
+                // String)
+                //                    Types.TIMESTAMP_WITH_TIMEZONE ->
+                //                        statement.setTimestamp(statementIndex, value as Timestamp)
+                //                    Types.TIME -> statement.setTime(statementIndex, value as Time)
+                //                    Types.TIMESTAMP -> statement.setTimestamp(statementIndex,
+                // value as Timestamp)
+                //                }
+            }
+        }
+        airbyteMetaStatementIndex?.let { statementIndex ->
+            statement.setString(statementIndex, Jsons.serialize(airbyteMeta))
+        }
+    }
+
+    fun readResult(rs: ResultSet, schema: List<NamedField>): ObjectValue {
+        val valueMap =
+            schema
+                .filter { field -> field.name !in airbyteFields }
+                .map { field ->
+                    try {
+                        when (field.type.type) {
```

Comment:
Same comment as above...this feels like another good use of an extension function on the type to encapsulate this conversion logic and make it easier to test.